main/main.c

```diff
@@ -659,7 +659,40 @@ PHPAPI ZEND_COLD void php_log_err_with_severity(char *log_message, int syslog_ty
 	if (PG(error_log) != NULL) {
 #ifdef HAVE_SYSLOG_H
 		if (!strcmp(PG(error_log), "syslog")) {
-			php_syslog(syslog_type_int, "%s", log_message);
+			static char nvt[] = " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
```

Comment:
@pprindeville thanks for the link. I've also checked the conversation on BT.

I think it's syslog-ng on my side as well, but anyway doing this <code>logger &#96;echo -e "x\ey"&#96;</code> writes `x#033y` in the log. So that implementation seems already to take care about the invalid bytes, while of course it can't be guaranteed for every logger implementation.

With regard to UTF-8 and others - the RFC paragraph does permit encodings other other than 7-bit. Most of encodings have lower 128 bytes compatible with ASCII, so replacing invalid bytes would likely not break anything, maybe some very exotic encodings. Another question is the method to choose for processing, that might get complicated and need some discussion. I would suggest to go by the full RFC compliance, as the relevant part is being touched anyway. The exact strategy should be probably discussed on internals@.

Thanks.